cratedb_toolkit/testing/testcontainers/cratedb.py

```diff
@@ -51,44 +57,91 @@ class CrateDBContainer(KeepaliveContainer, DbContainer):
     CRATEDB_PASSWORD = os.environ.get("CRATEDB_PASSWORD", "")
     CRATEDB_DB = os.environ.get("CRATEDB_DB", "doc")
     KEEPALIVE = asbool(os.environ.get("CRATEDB_KEEPALIVE", os.environ.get("TC_KEEPALIVE", False)))
+    CMD_OPTS = {
+        "discovery.type": "single-node",
+        "node.attr.storage": "hot",
```

Comment:
Yeah, well spotted. It is indeed a configuration detail which is specific to the test cases for `cratedb_toolkit.retention`. 

@pilosus: Do you think we can improve this spot, so that corresponding configuration settings can be defined on behalf of the snippet in `conftest.py`? This time, it will probably not be so easy, because the test adapter will already need this information at startup time.

Reply:
> This time, it will probably not be so easy, because the test adapter will already need this information at startup time. Maybe you have an idea how to handle this elegantly?
>
> While being at it: Of course, it would not just be about the specific `node.attr.storage` parameter, but about any other parameters as well.

Ah, I see there is already `cmd_opts` on the constructor. So, `{"node.attr.storage": "hot"}` would just need to be moved over there.

Reply:
Exactly. And given that we use [dict merge](https://github.com/crate-workbench/cratedb-toolkit/pull/82/files#diff-a269e42ec7e05ba66ed28ac184deeaa00fb81a0db3ccb47d34635c654f1b68bcR95), you can even override the default settings

```
In [1]: from cratedb_toolkit.testing.testcontainers.cratedb  import CrateDBContainer

In [2]: c = CrateDBContainer(cmd_opts={"node.attr.storage": "cold"})

In [3]: c._command
Out[3]: '-Cdiscovery.type=single-node -Cnode.attr.storage=cold -Cpath.repo=/tmp/snapshots'
```

I'll remove the `node.attr.storage` from the default though

Reply:
Thanks. What do you think about `path.repo`, @seut? It could be convenient for testing to have it configured by default.

Reply:
@seut @amotl hmm, we I delete `node.attr.storage=hot`
tests: 
- `tests/retention/test_cli.py::test_run_delete_basic` 
- `tests/retention/test_cli.py::test_run_delete_dryrun`
- `tests/retention/test_cli.py::test_run_reallocate`
hang.

Here are the logs:
```
2023-11-24 17:39:14,494 [cratedb_toolkit.retention.setup.schema] INFO    : Installing retention policy bookkeeping table at database 'crate://crate:REDACTED@localhost:33018', table TableAddress(schema='testdrive-ext', table='retention_policy')
2023-11-24 17:39:14,902 [cratedb_toolkit.retention.store     ] INFO    : Connecting to database crate://crate:REDACTED@localhost:33018, table "testdrive-ext"."retention_policy"
Waiting to be ready...
2023-11-24 17:39:15,366 [testcontainers.core.waiting_utils   ] INFO    : Waiting to be ready...
```

Interestingly enough, `test_run_delete_with_tags_match` works well

Reply:
Yeah, the test cases defined for `cratedb_toolkit.retention` need this setting to be configured, so it will need to go into `tests/conftest.py` somehow. On the other hand, it should not be part of the generic configuration. That's yet another cliff we need to take.

Reply:
Given the whole chain of the nested fixtures with different scopes, it will either require some more time from me next week, or a simpler solution with `cratedb` override for `tests/retention/conftest.py` that have `node.attr.storage=hot` in it.

Reply:
We can easily handle that on a later iteration, and/or discuss possible solutions beforehand.